utilities/four_c_utils/src/four_c_utils/check_includes.py

```diff
@@ -0,0 +1,120 @@
+# This file is part of 4C multiphysics licensed under the
+# GNU Lesser General Public License v3.0 or later.
+#
+# See the LICENSE.md file in the top-level for license information.
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+"""Check that C++ files have correct includes"""
+
+import re
+import sys
+
+from pathlib import Path
+from typing import Tuple
+
+from four_c_utils.common_utils import file_contents
+
+
+def replace_line_in_file(file: Path, line_no: int, line: str):
+    """Replace a line in a file.
+
+    Args:
+        file: The file to replace the line in
+        line_no: The line number to replace
+        line: The new line
+    """
+
+    lines = file_contents(file)
+
+    lines[line_no] = line
+
+    with open(file, "w") as f:
+        f.writelines(lines)
+
+
+def check_and_fix_includes(file: Path) -> Tuple[list, bool]:
+    """Check that the file has valid includes.
+
+    If quotes are missing for internal imports fix them.
+
+    Check that 4C files are included with quotes, external
+    ones are included with angle brackets. Overall, no relative
+    directory navigation (using '.' or '..') is allowed in
+    includes.
+
+    Args:
+        file: The file to check
+
+    Returns:
+        List of line numbers with invalid includes, bool if
+        includes were fixed
+    """
+
+    invalid_include_lines = []
+    fixed_include = False
+
+    for line_no, line in enumerate(file_contents(file)):
+
+        if not line.startswith("#include"):
+            continue
+
+        if ".." in line or "./" in line:
+            invalid_include_lines.append([line_no + 1, line])
+            continue
+
+        if "4C" in line:
+            match = re.search(r"#include <(4C_\w+\.hpp)>", line)
+            if match:
+                new_line = f'#include "{match[1]}"\n'
+                replace_line_in_file(file, line_no, new_line)
```

Comment:
I think this removes and comments followed by the include which is not necessarily wanted.

```suggestion
            match = re.search(r"#include <(4C_\w+\.hpp)>(.*)", line)
            if match:
                new_line = f'#include "{match[1]}"{match[2]}'
                replace_line_in_file(file, line_no, new_line)
```

Reply:
That's a good idea! Thanks for the specific suggestion - works flawlessly 🙏🏻